src/NativeShell/src/cordova/NativeShell.java

```diff
@@ -0,0 +1,188 @@
+            jsonObject.put("appVersion", appVersion);
+        } catch(Exception e) {
+            callbackContext.error("error: " + e.getMessage());
+            return false;
+        }
+
+        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, jsonObject);
+        callbackContext.sendPluginResult(pluginResult);
+        return true;
+    }
+
+    public boolean enableFullscreen() {
+        cordova.getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                int visibility = SYSTEM_UI_FLAG_FULLSCREEN
+                        | SYSTEM_UI_FLAG_IMMERSIVE_STICKY
+                        | SYSTEM_UI_FLAG_HIDE_NAVIGATION;
+                cordova.getActivity().getWindow().getDecorView().setSystemUiVisibility(visibility);
+                cordova.getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+                cordova.getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN);
+            }
+        });
+        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK);
+        callbackContext.sendPluginResult(pluginResult);
+        return true;
+    }
+
+    public boolean disableFullscreen() {
+        cordova.getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                cordova.getActivity().getWindow().getDecorView().setSystemUiVisibility(SYSTEM_UI_FLAG_VISIBLE);
+                cordova.getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN);
+                cordova.getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+            }
+        });
+        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK);
+        callbackContext.sendPluginResult(pluginResult);
+        return true;
+    }
+
+    public boolean updateMediaSession() {
+        String action;
+        String itemId;
+        String title;
+        String artist;
+        String album;
+        String imageUrl;
+
+        int duration;
+        int position;
+
+        boolean canSeek;
+        boolean isPaused;
+        boolean isLocalPlayer;
+        try {
+            JSONObject options = args.getJSONObject(0);
+            action = options.getString("action");
+            itemId = options.getString("itemId");
+            title = options.getString("title");
+            artist = options.getString("artist");
+            album = options.getString("album");
+            imageUrl = options.getString("imageUrl");
+
+            duration = options.getInt("duration");
+            position = options.getInt("position");
+
+            canSeek = options.getBoolean("canSeek");
+            isPaused = options.getBoolean("isPaused");
+            isLocalPlayer = options.getBoolean("isLocalPlayer");
+        } catch (Exception e) {
+            callbackContext.error("error: " + e.getMessage());
```

Comment:
can we show more specific error message here, maybe like `error parsing arguments for updateMediaSession:` ?

Reply:
how about `"error: cannot parse arguments for updateMediaSession: "`? just a tad bit more context for the log reader